include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -1126,35 +1156,58 @@ LAPACKFullMatrix<number>::Tvmult(Vector<number2> &,
 }
 
 
+
 template <typename number>
 template <typename number2>
 void
 LAPACKFullMatrix<number>::Tvmult_add(Vector<number2> &,
                                      const Vector<number2> &) const
 {
   Assert(false,
-         ExcMessage(
-           "LAPACKFullMatrix<number>::Tvmult_add must be called with a "
-           "matching Vector<double> vector type."));
+         ExcMessage("LAPACKFullMatrix<number>::Tvmult_add must be called "
+                    "with a matching Vector<double> vector type."));
 }
 
 
+
+namespace internal
+{
+  namespace LAPACKFullMatrixImplementation
+  {
+    template <typename RealNumber>
+    std::complex<RealNumber>
+    pack_complex(const RealNumber &real_part, const RealNumber &imaginary_part)
+    {
+      return std::complex<RealNumber>(real_part, imaginary_part);
+    }
+
+    // The eigenvalues in LAPACKFullMatrix with complex-valued matrices are
+    // contained in the 'wi' array, ignoring the 'wr' array.
+    template <typename Number>
+    std::complex<Number>
+    pack_complex(const Number &, const std::complex<Number> &complex_number)
+    {
+      return complex_number;
+    }
+  } // namespace LAPACKFullMatrixImplementation
+} // namespace internal
+
+
+
 template <typename number>
-inline std::complex<number>
+inline std::complex<typename numbers::NumberTraits<number>::real_type>
 LAPACKFullMatrix<number>::eigenvalue(const size_type i) const
 {
   Assert(state & LAPACKSupport::eigenvalues, ExcInvalidState());
   Assert(wr.size() == this->n_rows(), ExcInternalError());
   Assert(wi.size() == this->n_rows(), ExcInternalError());
   AssertIndexRange(i, this->n_rows());
 
-  if (numbers::NumberTraits<number>::is_complex)
-    return std::complex<number>(wi[i]);
-  else
-    return std::complex<number>(wr[i], wi[i]);
+  return internal::LAPACKFullMatrixImplementation::pack_complex(wr[i], wi[i]);
```

Comment:
Without `if constexpr`, I need to call into different helper functions as either one of the two versions would not compile.